projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -310,6 +310,10 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
     }
 
     public filter_global(term, condition, ignoreCase) {
+        if (!condition) {
```

Comment:
It doesn't seem to me that this is the best solution. A few lined down the same method, the condition is checked for being defined, and if not, then no new filtering expression is added. I think the issue comes with the filtering operand reset which happens before the check is performed. If you move it inside the check, then there should be no error.